lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/openturns/FunctionalChaosValidation.hxx

```diff
@@ -0,0 +1,97 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The validation of a chaos expansion
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_FUNCTIONALCHAOSVALIDATION_HXX
+#define OPENTURNS_FUNCTIONALCHAOSVALIDATION_HXX
+
+#include "openturns/Sample.hxx"
+#include "openturns/FunctionalChaosResult.hxx"
+#include "openturns/MetaModelValidation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class FunctionalChaosValidation
+ *
+ * The validation of a chaos expansion
+ */
+
+class OT_API FunctionalChaosValidation
+  : public MetaModelValidation
+{
+  CLASSNAME
+
+public:
+  typedef Collection<Function>           FunctionCollection;
+  enum CrossValidationMethod {LEAVEONEOUT = 0, KFOLD = 1};
+
+  /** Default constructor */
+  FunctionalChaosValidation();
+
+  /** Parameter constructor */
+  FunctionalChaosValidation(const FunctionalChaosResult & chaosResult,
+                            const CrossValidationMethod method, 
+                            const UnsignedInteger & kParameter = ResourceMap::GetAsUnsignedInteger("FunctionalChaosValidation-DefaultKFoldParameter"));
```

Comment:
we should use the KFold/LeaveOneOut objects here instead of an enum, and kParameter should from the KFold objects getK method

Reply:
I agree. Even more: we could as well provide a ctor which takes a FunctionalChaosAlgorithm and a splitter. This way, the validation can estimate the error even in case of a model selection.

I have a draft of this idea in 060807722d56808ad772d342ed91d34dd01b4e60. I continue to work on it.

Reply:
@jschueller : In b0db1bff5be5477cd06e38e4b877a65c43f5cee0, you will find a working implementation of this for the linear model. If you validate this proposal, I will reflect the changes to the FunctionalChaosValidation.

Reply:
I would rather use the Splitter interface class instead of two constructors, even if casts should be made

Reply:
I followed that path in the first place: one single ctor with a `SplitterImplementation`. But then I have to distinguish between `LeaveOneOutSplitter` and `KFoldSplitter` because the 2 implementations are different: how to see which is which afterwards?

Reply:
you can get the class name

Reply:
@jschueller : The class name of a `SplitterImplementation` class is "SplitterImplementation" which does not help much. The only way I see is to dynamically cast into a `LeaveOneOutSplitter`. If it fails, this means that this is a `KFoldSplitter`. It does not seem to be easy to maintain to me.

Reply:
you can use "getImplementation()->getClassName()" from the Splitter class (not SplitterImplementation)

Reply:
I could use this if there was a `Splitter` class, but there is no one:
![image](https://github.com/openturns/openturns/assets/31351465/fe5dc1e6-4d1d-443c-9095-297372998aed)

Reply:
oh :{

Reply:
- Create a `Splitter` class.
- See if we can implement new methods of the `Splitter` class that implements the service we need for cross-validation.
- See if we can create two different cross-validation classes: one for LOO, one for K-Fold. Both can derive from a mother class.